sequencing/ATACSeq/preprocessing/fastp.sh

```diff
@@ -0,0 +1,61 @@
+## - execute from scripts directory                                                                                   ||
+##                                                                                                                    ||
+## DESCRIPTION: This script performs trimming of raw reads and following quality control of the new trimmed reads.    ||
+##                                                                                                                    ||
+## REQUIRES:                                                                                                          ||
+## - Variables in config file: RAWDATADIR, TRIMDIR                                                                    ||
+## - Software: fastp, fastqc                                                                                          ||
+##                                                                                                                    ||
+## INPUTS:                                                                                                            || 
+## $1 -> <sampleName> Name of sample specified in command line.                                                       ||
+## $2 -> <R1> Read 1 name                                                                                             ||
+## $3 -> <R2> Read 2 name                                                                                             ||
+##                                                                                                                    ||
+## OUTPUTS:                                                                                                           || 
+##  *_fastp.html, *_trimmed.f, *_fastp.json                                                                           ||
+##                                                                                                                    ||
+## ===================================================================================================================##
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
+
+
+sampleName=$1
+
+echo
+echo "Starting trimming on" ${sampleName} "at: "
+date -u
+
+## Get read 1 and read 2 for specified sample name
+f1=$(basename $2)
+f2=$(basename $3)
+
+cd ${RAWDATADIR}
+
+## create output filenames
+outf1=${f1/.f/_trimmed.f}
+outf2=${f2/.f/_trimmed.f}
+
+mkdir -p ${TRIMDIR}/qc/
+
+##Specific adapters for R1 and R2 need to be specified for correct trimming
+adapterR1=CTGTCTCTTATACACATCTCCGAGCCCACGAGAC
+adapterR2=CTGTCTCTTATACACATCTGACGCTGCCGACGA 
+
+## ============ ##
+##    TRIM      ##
+## ============ ##
+
+fastp --detect_adapter_for_pe --adapter_sequence=$adapterR1 --adapter_sequence_r2=$adapterR2 --length_required=27 --thread=8 --in1=${f1} --in2=${f2} --out1=${TRIMDIR}/${outf1} --out2=${TRIMDIR}/${outf2} --html=${TRIMDIR}/qc/${sampleName}_fastp.html --json=${TRIMDIR}/qc/${sampleName}_fastp.json
```

Comment:
Why are you specifying the worker thread number here (--thread=8)? Considering you are parallelising the trimming process, you could base the worker thread off of the array ID for a speed up.

So you could use modular arithmetic here right? Array task 1 goes to thread 1, task 2 to thread 2, ..., task 16 to thread 16, task 17 back to thread 1 etc.

This way you can take advantage of running the job in parallel, instead of running everything on the same thread.

Reply:
```bash
--thread=$(( (${SLURM_ARRAY_TASK_ID} % 16) + 1 ))
```

The `+ 1` here is so that you don't get `--thread=0` but instead `--thread=1-16`. I think you'd need to parse in the $SLURM_ARRAY_TASK_ID into this script as I'm not sure what the scope of this variable is.